components/camel-kafka/src/test/java/org/apache/camel/component/kafka/integration/KafkaBreakOnFirstErrorSeekIssueIT.java

```diff
@@ -0,0 +1,146 @@
+package org.apache.camel.component.kafka.integration;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.kafka.MockConsumerInterceptor;
+import org.apache.camel.component.kafka.testutil.CamelKafkaUtil;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * this will test breakOnFirstError functionality
+ * and the issue that was surfaced in CAMEL-19894
+ * regarding failure to correctly commit the offset
+ * in a batch using the Synch Commit Manager
+ * 
+ *  mimics the reproduction of the problem in 
+ *  https://github.com/Krivda/camel-bug-reproduction
+ */
+class KafkaBreakOnFirstErrorSeekIssueIT extends BaseEmbeddedKafkaTestSupport {
+
```

Comment:
I was adding integration tests for this fix
There are 3 that are basic and show how breakOnFirstError will work

Still working on adding IT for the 2 issues
Note: need to create multiple partitions in the BaseEmbeddedKafkaTestSupport 
That doesn't seem to work


The 2 basic tests and the 2 tests that were part of reproducing the issues in [CAMEL-20044](https://issues.apache.org/jira/browse/CAMEL-20044) and [CAMEL-19894](https://issues.apache.org/jira/browse/CAMEL-19894) are captured in this repo

https://github.com/CodeSmell/CamelKafkaOffset

Reply:
At this point, they should still fail. Right? I noticed this one failed on my manual tests ...

Reply:
That's correct. This test will fail as-is. 
I have not had a chance to look at the infra test to get the partitions increased. 
The same test is in my personal repo (link above). That was passing when I ran it with my local build of the camel-kafka component with these changes.